wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -588,6 +611,149 @@ private String inferEngine() throws ModelNotFoundException {
+                || Files.isRegularFile(modelDir.resolve("config.json"));
+    }
+
+    private String inferLMIEngine() throws ModelException, IOException {
+        // MMS/Torchserve
+        if (Files.isDirectory(modelDir.resolve("MAR-INF"))) {
+            return "Python";
+        }
+        HuggingFaceModelConfig modelConfig = getHuggingFaceModelConfig();
+        if (modelConfig == null) {
+            return "Python";
+        }
+
+        int tensorParallelDegree = CudaUtils.getGpuCount();
+        if (Utils.getEnvOrSystemProperty("TENSOR_PARALLEL_DEGREE") != null) {
+            tensorParallelDegree =
+                    Integer.parseInt(Utils.getEnvOrSystemProperty("TENSOR_PARALLEL_DEGREE"));
+        } else if (prop.get("option.tensor_parallel_degree") != null) {
+            tensorParallelDegree =
+                    Integer.parseInt(prop.get("option.tensor_parallel_degree").toString());
+        }
+
+        if (tensorParallelDegree > 0) {
+            prop.put("option.tensor_parallel_degree", tensorParallelDegree);
+        }
+        String huggingFaceTask = Utils.getEnvOrSystemProperty("HF_TASK");
+        if (huggingFaceTask != null) {
+            prop.put("option.task", huggingFaceTask);
+        }
+
+        if ("stable-diffusion".equals(modelConfig.getModelType())) {
+            // TODO: Move this from hardcoded to deduced in PyModel
+            prop.put("option.entryPoint", "djl_python.stable-diffusion");
+            return "DeepSpeed";
+        }
+
+        if (!isTensorParallelSupported(modelConfig.getNumAttentionHeads(), tensorParallelDegree)) {
+            return "Python";
+        }
+        if (isDeepSpeedRecommended(modelConfig.getModelType())) {
+            return "DeepSpeed";
+        }
+        if (isFasterTransformerRecommended(modelConfig.getModelType())) {
+            return "FasterTransformer";
+        }
+        return "Python";
+    }
+
+    private URI generateHuggingFaceConfigUri(String modelId) throws ModelException, IOException {
+        URI configUri = null;
+        if (modelId != null && modelId.startsWith("s3://")) {
+            // This is definitely suboptimal, but for the majority of cases we need to download this
+            // s3 model eventually, so it is not the worst thing to download it now.
+            downloadS3();
+            if (Files.isRegularFile(downloadS3Dir.resolve("config.json"))) {
+                configUri = downloadS3Dir.resolve("config.json").toUri();
+            } else if (Files.isRegularFile(downloadS3Dir.resolve("model_index.json"))) {
+                configUri = downloadS3Dir.resolve("model_index.json").toUri();
+            }
+        } else if (modelId != null) {
+            prop.put("option.modelId", modelId);
+            configUri = URI.create("https://huggingface.co/" + modelId + "/raw/main/config.json");
+            HttpURLConnection configUrl = (HttpURLConnection) configUri.toURL().openConnection();
```

Comment:
We can consider use "OPTION" method instead of GET. I think it should work

Reply:
I'll explore that and add as follow up if i get it to work.